realm/src/main/java/io/realm/Realm.java

```diff
@@ -123,25 +122,33 @@
 public final class Realm implements Closeable {
     public static final String DEFAULT_REALM_NAME = "default.realm";
 
-    protected static final ThreadLocal<Map<String, Realm>> realmsCache = new ThreadLocal<Map<String, Realm>>() {
-        @SuppressLint("UseSparseArrays")
```

Comment:
No, SparseArrays are only relevant if the key is an Integer, which we removed some time ago.